compiler/runtime/OMRRuntimeAssumptions.hpp

```diff
@@ -185,6 +187,59 @@ class PatchNOPedGuardSite : public OMR::LocationRedirectRuntimeAssumption
    uint8_t *_destination;
    }; // TR::PatchNOPedGuardSite
 
+class PatchSites
+   {
+   private:
+   int       _refCount;
+   int       _size; 
+   uint8_t **_patchPoints;
```

Comment:
The choice of the location and destination being in the same array is interesting - what is the reasoning behind this as opposed to having two arrays of the same length - one for location and one for destination. Is it for locality of data at runtime? If so a comment would be appropriate.

Reply:
It improves locality and reduces the number of allocations that are necessary. A single allocation could achieve the same effect, with separate points for the first and second halves of the list, but this would waste an additional field on the class. I have added a comment that expresses this.